src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -296,3 +274,19 @@ export default function connectHierarchicalMenu(renderFn, unmountFn = noop) {
     };
   };
 }
+
+function formatFacetValues(facetValues, currentLimit) {
+  return facetValues
+    .slice(0, currentLimit)
```

Comment:
That'd be optimal. What API would you see?

Reply:
hmm, the recursiveness in hierarchical indeed poses a problem. That means that you can not really do much more than wrap the slice? Maybe someone else has an idea

Reply:
> hmm, the recursiveness in hierarchical indeed poses a problem. That means that you can not really do much more than wrap the slice? Maybe someone else has an idea

can you elaborate the problem? I don't understand how it can be a problem.